src/tests/ftest/util/general_utils.py

```diff
@@ -44,6 +44,60 @@ class DaosTestError(Exception):
     """DAOS API exception class."""
 
 
+def clush(hostlist, command, env=None, verbose=True):
+    """Run a command in parallel on a list of nodes using clush.
+
+    Args:
+        hostlist (list): list of nodes on which to run the command in parallel
+        command (str): the command to run in parallel
+        env (dict, optional): dictionary of environment variables to set.
+            Defaults to None.
+        verbose (bool, optional): should the command output be displayed.
+            Defaults to True.
+
+    Returns:
+        CmdResult: a CmdResult object containing the result of the command:
+            .command        command issued
+            .exit_status    return code from the command
+            .stdout         raw stdout (bytes)
+            .stderr         raw stderr (bytes)
+            .duration       duration of the command
+            .interrupted    was the command interrupted
+            .pid            pid of the command
+
+    """
+    nodeset = NodeSet.fromlist(hostlist)
+    clush_command = "clush -w {} -B -S \"{}\"".format(nodeset, command)
+    return process.run(
+        clush_command, env=env, ignore_status=True, shell=True,
+        verbose=verbose, allow_output_check="none")
```

Comment:
Given that _ClusterShell_ is primarily a python library, shouldn't we avoid the trip through a `fork()`/`exec()` and the shell here and work with _ClusterShell_ directly like you did for `check_file_exists()`?

Reply:
Yes.  I've updated the method and in doing so discovered that using the maximum return code is problematic with commands like pkill and pgrep.  The new method returns a dictionary of each return code detected - paired with the nodes that returned the code - to address this issue.